lib/internal/fs/promises.js

```diff
@@ -346,7 +346,16 @@ async function open(path, flags, mode) {
                                  flagsNumber, mode, kUsePromises));
 }
 
-async function read(handle, buffer, offset, length, position) {
+async function read(handle, bufferOrOptions, offset, length, position) {
+  let buffer = bufferOrOptions;
+
+  if (!isArrayBufferView(buffer)) {
```

Comment:
we check `isArrayBufferView` twice, the second is `validateBuffer` function

Reply:
> we check isArrayBufferView twice, the second is validateBuffer function

I've used it as a mechanism to check if buffer is truely `<Buffer> | <TypedArray> | <DataView>`<sup>[[1]](https://nodejs.org/api/fs.html#fs_fs_read_fd_buffer_offset_length_position_callback)</sup> of if it's just object containing read options, since `typeof buffer` is `'object'` in both cases. And `buffer instanceof Object` also returns `true` whether buffer is truely `Buffer` or regular `Object`

Then again, I could use some thing like:
```suggestion
  if ( !([Buffer, TypedArray, DataView].some(c => buffer instanceof c)) ) {
```
but I think this would be essentialy the same thing, and I feel more comfortable using tested functions inside NodeJS. Plus, if parameters change in the future, this option would then require changing.

Is there any _easy/direct/maintainable_ way of checking of the argument is `Object` or `<Buffer> | <TypedArray> | <DataView>`? Maybe using `Object.prototype`?

Reply:
Suggestion: https://github.com/nodejs/node/pull/34180/files#r523406882